examples/preprocessing/plot_xdawn_denoising.py

```diff
@@ -34,7 +34,7 @@
 from mne import (io, compute_raw_covariance, read_events, pick_types,
                  Epochs)
 from mne.datasets import sample
-from mne.preprocessing import Xdawn
+from mne.preprocessing import XdawnTransformer
```

Comment:
Good point. We're indeed facing 2 distinct issues that I mixed in this PR:

1. The XdawnTransformer did not take overlapping events. 

I changed the code to have an `events` param at fit time: `xd.fit(X, y, events=events)`, but I'm wondering whether this is such a great idea. It's sklearn compatible but not conventional (yet?) . So, there is a risk that this API is not going to be sustainable.

2. The Xdawn class is an API mess. 

I modularized its core computation into separate private functions. 

The XdawnTransformer now does the *very* same computations but keeps a strict and simple API closer to sklearn than to `mne.ICA` or `mne.GAT`. Specifically, it doesn't have an `apply` method, takes numpy arrays, does not keep `mne.Evoked` objects in its attributes etc.